### ISSUE TITLE
Import Assets: Create and Update not working

### ISSUE THREAD

**Issue by bjhiggins** [tracmor/tracmor#85]
I have just upgraded my development environment to the latest master.zip from GitHub (b20edffd82).

Scenario: I want to import a list of 1000 assets with "To Be Received" for location, add them to a shipping receipt, then as they are delivered out to final locations, reimport the same CSV with the actual location names for location.

Step 1: Initial Import.
Click Import Assets, select your CSV file, check header row, import action set to Create Records.  
Click next, make sure that all default and custom required fields are mapped (we have a serial number and asset status custom required fields).
Click import now, and my assets are properly imported, all marked To Be Received.

Step 2: Update Import
Click Import Assets, select your CSV file, check header row, import action set to Create and Update Records.  
Click next, make sure that all default and custom required fields are mapped.  Click Import now.

Tracmor indicates "You must select all required fields (Serial Number, Asset Status, Asset Tag, Model Short Description and Location)."  The field mapped to Serial Number has a message "You must enter default value." - however the Default Value field has a 0 in it.

If I change the 0 to 123456 and click Import Now, I'm brought back to the same screen with the same general error message, however there is no specific error message on Serial Number.

If I rerun my Step 2 and change 0 to 123456 - I receive the same general error message, but no indication as to what is wrong.

I can find no PHP or other suspect errors in the Apache/PHP logs.

Thoughts?

**Comment by jonwitts** [comment]
When you are updating assets you need to first export the assets from Tracmor so that it includes the asset ID which Tracmor assigns it when you first import them...

**Comment by bjhiggins** [comment]
When updating an asset, what are required fields - just the asset id?

**Comment by jonwitts** [comment]
The process I follow is along these lines...
Upload assets from csv file and match all custom fields etc. over.
Update the fields displayed on asset search so that I have a list of the assets I just uploaded, then download them as csv. 
Edit the csv (taking care that leading zeros and date formats are left alone) as needed.
Upload the edited csv this time choosing "Create and Update"....

Kinda worked it out through trial and error as the docs were not too clear on this part...

**Comment by bjhiggins** [comment]
I tried exporting 6 imported assets (say that three times fast), then updated the location on them.  When I re-imported with , I got to the Import Results and none of the records were imported.  No error or reason given.

Thoughts?

**Comment by jonwitts** [comment]
Does the updated location match an existing location in your setup exactly?

**Comment by bjhiggins** [comment]
Yes - and in a previous import, the location string worked without problem.  So I don't think it's the location string I'm using.

**Comment by bjhiggins** [comment]
Any thoughts as to what is happening or where to look for the errors?

**Comment by bjhiggins** [comment]
Think I figured it out... You can import an Asset to location "To Be Received" - but once it's in that location, you cannot update it via an import.  Once I clicked received on the receipt, I was able to update the record...

The problem is that I need to import 1168 assets, add them to a receipt, then receive them to appropriate locations.  I really don't want to click on 1168 drop-down controls to choose a location and then click on receive for each and every one of them...  :(

**Comment by bjhiggins** [comment]
Solved the issue by writing three SQL queries for each asset with I imported.  One to update the location on the asset, one to insert an asset_transaction record to tie a move from To Be received to received to the receipt, one to insert an asset_transaction record to tie to a transaction to move from the loading dock to the final location.

Cumbersome... but doable...
